### ISSUE TITLE
Sandbox `host-log-level` fails if no `VERBOSE_LOGGING` set

### ISSUE THREAD

**Issue by rengolin** [microsoft/CCF#3753]
When running the sandbox with `-l trace` I get a cryptic stack trace on the Python log output. Investigating the error log, I see:
```
terminating with uncaught exception of type std::logic_error: Error parsing configuration file .../ccf/build/workspace/sandbox_common/0.config.json: "Trace" is not convertible to Level
```
Turns out I hadn't built it with `-DVERBOSE_LOGGING=True`.

While that saves some code in the C++ side, it doesn't help that `start_network.py` still lists `trace` and `debug` as valid options and create the config file which will subsequently crash on `host/main.cpp`:
```
config = nlohmann::json::parse(config_str);
```
and throw that exception.

I'm not sure what can be done, because checking if the CMake option is active isn't really a doable thing from a Python script that doesn't know where the build directory is, but perhaps a warning if the user chooses either `trace` or `debug` that the host may crash if the option isn't on (even when it is)?

**Comment by jumaffre** [comment]
Ideally, the host logging level should be able to be set to any value (including `trace` and `debug`) even if `cchost` wasn't build with `-DVERBOSE_LOGGING` (since the host is untrusted anyway). I think this is the real fix here.

The issue is that the host and enclave `logger` share the same definition of `logger::Level`. You may think that it'd be as easy as creating two new enum definitions but it's not as simple since the enclave passes its log messages (along with a level) to the host to be printed to `stdout`. It's definitely doable but not easy enough that one of us has ever picked it up. 

We can now use this issue to remind us to do it at some point!

**Comment by jumaffre** [comment]
Fixed by #3762
